pyproject.toml

```diff
@@ -41,8 +41,10 @@ requires-python = ">=3.7"
 dependencies = ["cryptography", "ipaddress"]
 
 [project.optional-dependencies]
-dev = ["pytest", "flask", "requests"]
-test = ["pytest"]
+dev = ["pytest", "flask", "requests", "pre-commit", "black", "ruff"]
```

Comment:
black isn't used here anymore, and technically ruff isn't either, because pre-commit installs it in its own env, if you want to remove one or both.

Reply:
I'll remove `black` for sure. I like having `ruff` installed locally during dev since I use [vim-ale](https://github.com/dense-analysis/ale) for in-editor feedback.